### PR TITLE
Add py2cfg

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
@conda-forge/help-python

**Comment by jan-janssen** [comment]
The windows tests fail, still I would prefer this package to remain `noarch`:
```
import: 'py2cfg'
No broken requirements found.
Traceback (most recent call last):
  File "C:\bld\py2cfg_1718453303736\_test_env\Scripts\py2cfg-script.py", line 5, in <module>
    from py2cfg._runner import main
  File "C:\bld\py2cfg_1718453303736\_test_env\lib\site-packages\py2cfg\_runner.py", line 10, in <module>
    import pty
  File "C:\bld\py2cfg_1718453303736\_test_env\lib\pty.py", line 12, in <module>
    import tty
  File "C:\bld\py2cfg_1718453303736\_test_env\lib\tty.py", line 5, in <module>
    from termios import *
ModuleNotFoundError: No module named 'termios'
```

**Review by xhochy — APPROVED** [review]

File: recipes/py2cfg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - pip
+  run:
+    - python >=3.7
+    - astor >=0.8.0,<0.9.0
+    - pudb >=2022.0.0,<2023.0.0
+    - websocket-server >=0.4
+    - python-graphviz >=0.1
+    - flask >=1.0.0,<2.0.0
+    - requests >=2.0.0,<3.0.0
+    - packaging
```
Comment (xhochy) [inline recipes/py2cfg/meta.yaml:35]:
```suggestion
    - packaging
    - __unix
```

**Review by xhochy — CHANGES_REQUESTED** [review]
It can stay noarch but it should not be installable on Windows

**Comment by jan-janssen** [comment]
@xhochy Thanks a lot, I updated the pull request accordingly

**Review by xhochy — APPROVED** [review]

**Pull request merged**